### PR TITLE
Onl 4560 - Migrate donut component

### PR DESCRIPTION
**Pull request by stavros-tomas**
Donut component migration to tailwind

### CODE REVIEW

File: src/components/ec-donut/ec-donut.vue [hunk 1]
```diff
@@ -1,6 +1,12 @@
 <template>
-  <div class="ec-donut">
-    <div class="ec-donut__container">
+  <div
+    class="ec-donut"
+    :data-test="$attrs['data-test'] ? `${$attrs['data-test']} ec-donut` : 'ec-donut'"
+  >
+    <div
+      class="ec-donut__container"
+      data-test="ec-donut-container"
```
Comment (mcibique) [inline src/components/ec-donut/ec-donut.vue:8]: all data-tests follows BEM and class name except this one.

File: src/components/ec-donut/ec-donut.vue [hunk 2]
```diff
@@ -101,92 +116,92 @@ export default {
...
-<style lang="scss">
-@import '../../scss/settings/colors/index';
-@import '../../scss/tools/';
-
+<style>
 .ec-donut {
-  display: flex;
-  flex-wrap: wrap;
+  @apply tw-flex;
+  @apply tw-flex-wrap;
```
Comment (mcibique) [inline src/components/ec-donut/ec-donut.vue:122]: You can join similar classes:
```
@apply tw-flex tw-flex-wrap;
```

File: src/components/ec-donut/ec-donut.vue [hunk 3]
```diff
@@ -101,92 +116,92 @@ export default {
...
-    @include media__up-to-sm {
-      width: 100%;
-      margin-right: 0;
-      display: flex;
-      justify-content: center;
+    @media screen and (max-width: theme('screens.sm')) {
+      @apply tw-w-full;
+      @apply tw-mr-0;
+      @apply tw-flex;
+      @apply tw-justify-center;
```
Comment (mcibique) [inline src/components/ec-donut/ec-donut.vue:133]: Same here:
```
@apply tw-flex tw-justify-center;
```

File: src/components/ec-donut/ec-donut.vue [hunk 4]
```diff
@@ -101,92 +116,92 @@ export default {
...
+    @apply tw-mr-24;
+
     height: 108px;
 
-    @include media__up-to-sm {
-      width: 100%;
-      margin-right: 0;
-      display: flex;
-      justify-content: center;
+    @media screen and (max-width: theme('screens.sm')) {
```
Comment (mcibique) [inline src/components/ec-donut/ec-donut.vue:129]: We are not following mobile first. Don't do max-width in `@media`

Make those CSS properties default and write a new CSS what should happen when screen is bigger than `sm`
```
&__container {
  @apply tw-w-full;
  @apply tw-mr-0;
  ...

  @screen sm { /* or md, I don't know */
     @apply tw-block;
     ...
  }
}
```

File: src/components/ec-donut/ec-donut.vue [hunk 5]
```diff
@@ -101,92 +116,92 @@ export default {
...
-      padding: 16px 0 0;
-      display: inline-flex;
-      flex-direction: column;
-      margin: 0 auto;
+    @apply tw-h4;
+    @apply tw-flex-wrap;
+    @apply tw-py-16;
+    @apply tw-text-gray-3;
+
+    @media screen and (max-width: theme('screens.sm')) {
```
Comment (mcibique) [inline src/components/ec-donut/ec-donut.vue:143]: Same problem here, use mobile first.

File: src/components/ec-donut/ec-donut.vue [hunk 6]
```diff
@@ -101,92 +116,92 @@ export default {
...
     }
   }
 
   &__legend-used {
-    margin-bottom: 24px;
+    @apply tw-mb-24;
 
-    @include media__up-to-sm {
-      margin-bottom: 12px;
+    @media screen and (max-width: theme('screens.sm')) {
```
Comment (mcibique) [inline src/components/ec-donut/ec-donut.vue:157]: Mobile first.

File: src/components/ec-donut/ec-donut.vue [hunk 7]
```diff
@@ -101,92 +116,92 @@ export default {
...
   }
 
   &__svg {
-    transform: rotate(-90deg);
+    @apply tw--rotate-90;
   }
 
   &__background {
-    stroke: $level-7-backgrounds;
+    @apply tw-bg-key-7;
```
Comment (mcibique) [inline src/components/ec-donut/ec-donut.vue:186]: 1. This changed stroke to background
2. It should be `gray-7` not `key-7`

File: src/components/ec-donut/ec-donut.vue [hunk 8]
```diff
@@ -101,92 +116,92 @@ export default {
...
-      display: inline-flex;
-      flex-direction: column;
-      margin: 0 auto;
+    @apply tw-h4;
+    @apply tw-flex-wrap;
+    @apply tw-py-16;
+    @apply tw-text-gray-3;
+
+    @media screen and (max-width: theme('screens.sm')) {
+      @apply tw-pt-16;
```
Comment (jaimeperezpedra) [inline src/components/ec-donut/ec-donut.vue:144]: Paddings can be on the same line
Margins can be on the same line
Flex classes can be on the same line

**Pull request merged**